docs/development/local_setup.md

```diff
@@ -1,67 +1,81 @@
 
-## Installing Golang environment
+This guide is split into three main parts:
+* Preparing your setup by installing all dependencies and tools
+* Building and starting Gardener components locally
+* Using your local Gardener setup to create a Shoot
 
-Install latest version of Golang. For MacOS you could use [Homebrew](https://brew.sh/):
+## Limitations of the local development setup
 
-```bash
-brew install go
-```
+You can run Gardener (API server, controller manager, scheduler, gardenlet) against any local Kubernetes cluster, however, your seed and shoot clusters must be deployed to a cloud provider.
+Currently, it is not possible to run Gardener entirely isolated from any cloud provider. This means that to be able create Shoot clusters you need to register an external Seed cluster (e.g., one created in AWS).
 
-For other OS, please check [Go installation documentation](https://golang.org/doc/install).
+# Preparing the Setup
 
-## Installing kubectl and helm
+## [macOS only] Installing homebrew
 
-As already mentioned in the introduction, the communication with the Gardener happens via the Kubernetes (Garden) cluster it is targeting. To interact with that cluster, you need to install `kubectl`. Please make sure that the version of `kubectl` is at least `v1.11.x`.
+The copy-paste instructions in this guide are designed for macOS and use the package manager [Homebrew](https://brew.sh/).
 
-On MacOS run
+On macOS run
+```bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+```
+
+## Installing git
+
+We use `git` as VCS which you need to install. On macOS run
 
 ```bash
-brew install kubernetes-cli
+brew install git
 ```
 
-Please check the [kubectl installation documentation](https://kubernetes.io/docs/tasks/tools/install-kubectl/) for other OS.
+On other OS, please check the [Git installation documentation](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git).
 
-You may also need to develop Helm charts or interact with Tiller using the [Helm](https://github.com/kubernetes/helm) CLI:
+## Installing Go
 
-On MacOS run
+Install the latest version of Go. On macOS run
 
 ```bash
-brew install kubernetes-helm
+brew install go
 ```
 
-On other OS please check the [Helm installation documentation](https://helm.sh/docs/intro/install/).
+For other OS, please check [Go installation documentation](https://golang.org/doc/install).
 
-## Installing git
+## Installing kubectl
 
-We use `git` as VCS which you need to install.
+Install `kubectl`. Please make sure that the version of `kubectl` is at least `v1.11.x`. On macOS run
+
+```bash
+brew install kubernetes-cli
+```
+
+Please check the [kubectl installation documentation](https://kubernetes.io/docs/tasks/tools/install-kubectl/) for other OS.
 
-On MacOS run
+## Installing helm
+You also need the [Helm](https://github.com/kubernetes/helm) CLI. On macOS run
```

Comment:
Should we state that it must be helm v3?

Reply:
I could add this, to be extra-safe. Although I think helm 2 has been out of maintenance since November 2020 and the instructions on the website and github release page (which be link to here) don't even mention helm 2, except for the 'are you upgrading from 2 to 3?' question on the website. So probably we're fine without this?

Reply:
When this documentation was written there was no helm 3, so implicitly it was helm 2 :)
But as we have already migrated to helm3 with https://github.com/gardener/gardener/pull/3994/ in [install-requirements.sh](https://github.com/gardener/gardener/blob/master/hack/install-requirements.sh), and helm3 is the default one in the community, I think it is fine as it is.